### ISSUE TITLE
How to determine the shape of input and output nodes in graph?  (C++) 

### ISSUE THREAD

**Issue by Pin80** [tensorflow/tensorflow#32493]
**System information**
- TensorFlow version (you are using): 1.14
- Operation system: Ubuntu 18.04
- Are you willing to contribute it (Yes/No): No

**Describe the current behavior**

At first, I tried to implement the following pipeline:

- 1a) I converted the model from keras (.hdf5) to tensorflow (.pb) and saved using the `tf.saved_model.simple_save` function in Python.
- 2a) I loaded the model using `tensorflow::LoadSavedModel`
- 3a) I determined a shape using the following code:

```C++
const string export_dir = "/home/user/MyBuild/build_tftest2/models";
 SavedModelBundle bundle; RunOptions run_options;
status = LoadSavedModel(sopt, run_options, export_dir,
               {kSavedModelTagServe}, &bundle);
if (status.ok())
{
    GraphDef graph = bundle.meta_graph_def.graph_def();
    auto shape = graph.node().Get(0).attr().at("shape").shape();
    for (int i = 0; i < shape.dim_size(); i++) {
        std::cout << shape.dim(i).size()<<std::endl;
    }
}
```
Later I was wishing to use optimal model, therefore I used another pipeline:

 - 1b) I converted the model from keras (.hdf5) to tensorflow (.pb) using 
```python
frozen_graph = convert_variables_to_constants(session, input_graph_def,
output_names, freeze_var_names)
```
 and saved in Python using:

```python
tf.train.write_graph(frozen_graph, "model", "tf_model.pb", as_text=False)
```
- 2b) I optimized the model using the `transform_graph` utility from tensorflow package.
- 3b) I loaded my graph using the following code (C++):
```c++
status = ReadBinaryProto(Env::Default(), "/home/user/MySoftware/foreign code/netology_JN/Diplom/Models/optimized/optim_model.pb", &graph_def);
```
- 4b) I tried to determine a shape using 3a) and an exception was thrown:

```
[libprotobuf FATAL /usr/local/include/google/protobuf/map.h:1064]
CHECK failed: it != end(): key not found: shape terminate called after
throwing an instance of 'google::protobuf::FatalException' what():
CHECK failed: it != end(): key not found: shape 14:21:36: The program
has unexpectedly finished.
```

**Comment by oanush** [comment]
@Pin80 ,
Can you please provide a simple and standalone code to reproduce the issue?Thanks!

**Comment by Pin80** [comment]
I solved this issue.
Tensorflow utility transform_graph didn't remove all unused tensors, when i called command: 
./transform_graph \
--in_graph='/home/user/MySoftware/foreign code/netology_JN/Diplom/Models/Raw/frozen_model.pb' \
--out_graph='/home/user/MySoftware/foreign code/netology_JN/Diplom/Models/optimized/optim_model.pb' \
--inputs='input_1' \
--outputs='conv2d_19_2/Sigmoid' \
--transforms='
strip_unused_nodes(type=float, shape="-1,192,512,1")
remove_nodes(op=Identity, op=CheckNumerics)
fold_old_batch_norms
The first node of my graph is not input tensor therefore it don't have "shape" attribute.
I found input tensor as second tensor in my graph and problem was solved.

P.S. This my first expirience of writing issue to tensorflow. Do  i need to close it ?
